<a href="https://colab.research.google.com/github/AlvaroBecker/CodingDojoTareas-/blob/main/encontrar_la_altura_promedio_por_ol%C3%ADmpico_individual_por_a%C3%B1o.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
import pandas as pd


In [1]:
def format_drive_url(url):
    file_id = url.split('/')[-2]
    download_url = f'https://drive.google.com/uc?id={file_id}&export=download'
    return download_url

In [9]:
url=format_drive_url('https://drive.google.com/file/d/1SrEVf2bzs_K_tr83C3Fs7Tu1v-fuEW72/view')
df=pd.read_csv(url)

1. Carguen este conjunto de datos y explórenlo. Respondan las siguientes preguntas:

     a. ¿Cuántas filas hay?

     b. ¿Cuántas columnas hay?

     c. ¿A qué columnas le faltan datos? (¿Por qué creen que hay muchos valores nulos en la columna Medals (medallas)?

     d. ¿Cuántas entradas corresponden a la ciudad de Londres?

     e. ¿Cuántos años tiene el atleta más joven en nuestros datos de muestra?  Pista: usen min()

2. Creen filtros para encontrar:

     a. ¿Cuántos atletas que participaron en el deporte "Tug-Of-War" (tira y afloja) pertenecían en el equipo “Sweden” (sueco) en el conjunto de datos de muestra?

     b. ¿Cuántas medallas de oro se le entregaron a los atletas que tenían más de 40 años?

3. Determinen la altura promedio de:

     a. Todas las filas incluidas.

     b. Todas las filas incluidas, pero agrupadas por Event (evento).

     c. Bono: Este conjunto de datos contiene muchos atletas más de una vez (si compitieron en varios eventos y/o varios años). Pueden confirmarlo encontrarlo .value_counts en la columna ID.  Para la tarea del bono queremos que obtengan la altura promedio de todos los atletas, pero sin incluir algún atleta más de una vez.  Pista: Aprendimos cómo eliminar duplicados si CADA columna fuera idéntica, sin embargo, ahora tenemos que eliminar una fila cada vez que un ID se repite.  Para hacer eso, pueden usar un subconjunto de sus datos como un argumento en la función drop_duplicates.  También le pueden decir a Python cuales duplicados mantener usando el argumento keep.



In [10]:
df

ID   Age  Height         Team  NOC        Games  Year  Season  \
0           5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter   
1           5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter   
2           5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter   
3           5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter   
4           5  27.0   185.0  Netherlands  NED  1994 Winter  1994  Winter   
...       ...   ...     ...          ...  ...          ...   ...     ...   
40611  135567  28.0   183.0       Russia  RUS  2004 Summer  2004  Summer   
40612  135570  27.0   176.0       Poland  POL  2014 Winter  2014  Winter   
40613  135570  27.0   176.0       Poland  POL  2014 Winter  2014  Winter   
40614  135571  30.0   185.0       Poland  POL  1998 Winter  1998  Winter   
40615  135571  34.0   185.0       Poland  POL  2002 Winter  2002  Winter   

                 City          Sport  \
0             Calgary  Speed Skating   
1             Calgary  Speed Skating   
2         Albertville  Speed Skating   
3         Albertville  Speed Skating   
4         Lillehammer  Speed Skating   
...               ...            ...   
40611          Athina         Rowing   
40612           Sochi    Ski Jumping   
40613           Sochi    Ski Jumping   
40614          Nagano      Bobsleigh   
40615  Salt Lake City      Bobsleigh   

                                          Event Medal  
0              Speed Skating Women's 500 metres   NaN  
1            Speed Skating Women's 1,000 metres   NaN  
2              Speed Skating Women's 500 metres   NaN  
3            Speed Skating Women's 1,000 metres   NaN  
4              Speed Skating Women's 500 metres   NaN  
...                                         ...   ...  
40611    Rowing Men's Lightweight Coxless Fours   NaN  
40612  Ski Jumping Men's Large Hill, Individual   NaN  
40613        Ski Jumping Men's Large Hill, Team   NaN  
40614                      Bobsleigh Men's Four   NaN  
40615                      Bobsleigh Men's Four   NaN  

[40616 rows x 12 columns]

a. ¿Cuántas filas hay?
b. ¿Cuántas columnas hay?

In [26]:
df.shape 

(40616, 12)

Hay 40616 filas y 12 columnas

c. ¿A qué columnas le faltan datos? (¿Por qué creen que hay muchos valores nulos en la columna Medals (medallas)?

In [15]:
df.isnull().sum()

ID            0
Age        1473
Height     9001
Team          0
NOC           0
Games         0
Year          0
Season        0
City          0
Sport         0
Event         0
Medal     34699
dtype: int64

d. ¿Cuántas entradas corresponden a la ciudad de Londres?


In [33]:
df.loc[df['City']=='London']['City'].count()

3370

e. ¿Cuántos años tiene el atleta más joven en nuestros datos de muestra? Pista: usen min()


In [34]:
df['Age'].min()

11.0

 Creen filtros para encontrar:

a.¿Cuántos atletas que participaron en el deporte "Tug-Of-War" (tira y afloja) pertenecían en el equipo “Sweden” (sueco) en el conjunto de datos de muestra?

In [42]:
df.loc[(df['Sport']=="Tug-Of-War")&(df['Team']=='Sweden')].groupby('ID')['Sport'].count().sum()

6

b. ¿Cuántas medallas de oro se le entregaron a los atletas que tenían más de 40 años?

In [47]:
df.loc[(df['Age']>40) & (df['Medal']=='Gold')]['Medal'].count()

66

Determinen la altura promedio de:

a. Todas las filas incluidas.

b. Todas las filas incluidas, pero agrupadas por Event (evento).

c. Bono: Este conjunto de datos contiene muchos atletas más de una vez (si compitieron en varios eventos y/o varios años). Pueden confirmarlo encontrarlo .value_counts en la columna ID. Para la tarea del bono queremos que obtengan la altura promedio de todos los atletas, pero sin incluir algún atleta más de una vez. Pista: Aprendimos cómo eliminar duplicados si CADA columna fuera idéntica, sin embargo, ahora tenemos que eliminar una fila cada vez que un ID se repite. Para hacer eso, pueden usar un subconjunto de sus datos como un argumento en la función drop_duplicates. También le pueden decir a Python cuales duplicados mantener usando el argumento keep.

In [48]:
df['Height'].mean()

175.3811165585956

In [49]:
df.groupby('Event')['Height'].mean()

Event
Alpine Skiing Men's Combined                      178.547170
Alpine Skiing Men's Downhill                      178.211679
Alpine Skiing Men's Giant Slalom                  176.739394
Alpine Skiing Men's Slalom                        176.482143
Alpine Skiing Men's Super G                       179.850575
                                                     ...    
Wrestling Women's Flyweight, Freestyle            158.100000
Wrestling Women's Heavyweight, Freestyle          173.444444
Wrestling Women's Light-Heavyweight, Freestyle    170.000000
Wrestling Women's Lightweight, Freestyle          162.666667
Wrestling Women's Middleweight, Freestyle         164.533333
Name: Height, Length: 726, dtype: float64

In [50]:
df['ID'].value_counts()

119591    36
44875     32
76437     31
133279    30
21462     29
          ..
78467      1
78468      1
78470      1
78474      1
91368      1
Name: ID, Length: 20336, dtype: int64

In [55]:
df_id_unique = df.drop_duplicates(subset='ID', keep='last').reset_index(drop=True)
df_id_unique['Height'].mean()

176.31818479042698